### PR TITLE
BREAKING CHANGE: Change error event format and fix crash due to error event on websocket

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: docs/src/events.md.ejs [hunk 1]
```diff
@@ -24,12 +24,22 @@ This event is emitted when there is an error on the connection to the server tha
...
+* the message that caused the error for `badMessage`
+* the error object emitted for `websocket`
+* the parsed response for rippled errors
 
 ### Example
 
 ```javascript
-api.on('error', (errorCode, errorMessage) => {
+api.on('error', (errorCode, errorMessage, data) => {
   console.log(errorCode + ': ' + errorMessage);
```
Comment (lumberj) [inline docs/src/events.md.ejs]: Just a nit but you didn't log the `data`

Comment (clark800) [inline docs/src/events.md.ejs]: Yeah, that was intentional, didn't think it was necessary

**Comment by lumberj** [comment]
+2

**Comment by lumberj** [comment]
I don't think this is a breaking change though, it just increases the arity of the event handler, but javascript does not enforce function arity

**Comment by clark800** [comment]
The breaking change is that the second parameter of the badMessage event moved to the third parameter

**Comment by lumberj** [comment]
Got it

**Pull request merged**